Fix clusters

danhanson opened the pull request:
Vehicles and Clusters now require a cluster id to be created. I also removed subclusters and cluster updates/edits. When adding a vehicle or a commodity, it is also added to its specified cluster. All of a cluster's vehicles and commodities are in the cluster object received from a read cluster request, and deleting a cluster results in all of its vehicles and commodities to be deleted. I have not fixed the postgresql errors though. I also plan on adding more tests.

danhanson committed:
added read all method to CrudDao

danhanson committed:
added ClusterQueries trait

danhanson committed:
added java lists to cluster

danhanson committed:
made vehicle and commodity models require commodityId to create

danhanson committed:
added cluster support to the controllers

danhanson committed:
made websocketCrudController support resource

danhanson committed:
added cluster support to commodities and vehicles

danhanson committed:
removed support for cluster updates

Vehicles and commmodities may only be add, deleted, or removed from
clusters through their respective endpoints.

danhanson committed:
fixed tests, I also had to remove nullable=false constraints from
cluster join columns

danhanson committed:
fixed tests again, had to change commodity and vehicle test since they
both require clusters now

aj-michael commented on app/io/pathfinder/controllers/ClusterController.java:
```diff
@@ -48,37 +48,15 @@ public Result createCluster() {
         } catch (PersistenceException e) {
             e.printStackTrace();
-            Logger.error(String.format("Error saving cluster to the database: %s", e.getMessage()));
+            Logger.error(String.format("Error saving cluster to the database: %s", e));
```
This prints the stack trace twice now.

aj-michael commented on app/io/pathfinder/data/ClusterQueries.scala:
```diff
@@ -0,0 +1,9 @@
+
+import com.avaje.ebean.Model
+import io.pathfinder.models.Cluster
+
```
Please add comment explaining what this is.

aj-michael committed:
Added comments and increased readability.

I was going to comment on PR #34, but due to shortness of time I just made the fixes myself.

aj-michael commented:
Instead of bitching about nitpicks in the comments, I decided to fix them myself.

LGTM.
